python/py-onnx/Portfile

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           github 1.0
+
+github.setup        onnx onnx 1.6.0 v
+revision            0
+name                py-${github.project}
+categories-append   science
+fetch.type          git
+
+platforms           darwin
+supported_archs     noarch
+
+license             BSD
+
+maintainers         nomaintainer
+
+description         Open Neural Network Exchange
+
+long_description    Open Neural Network Exchange (ONNX) is an open\
+                    ecosystem that empowers AI developers to choose\
+                    the right tools as their project evolves. ONNX\
+                    provides an open source format for AI models, both\
+                    deep learning and traditional ML. It defines an\
+                    extensible computation graph model, as well as\
+                    definitions of built-in operators and standard\
+                    data types. Currently we focus on the capabilities\
+                    needed for inferencing (scoring).
+
+homepage            https://onnx.ai/
+
+# Support python versions.
+python.versions     27 35 36 37
+
+compiler.cxx_standard 2011
+
+if {${name} ne ${subport}} {
+
+    depends_build-append \
+        port:git \
+        port:cctools \
+        path:bin/cmake:cmake \
+        port:py${python.version}-protobuf3 \
+        port:py${python.version}-pytest-runner
+
+    depends_lib-append \
+        port:protobuf-c \
+        port:py${python.version}-future \
+        port:py${python.version}-numpy \
+        port:py${python.version}-setuptools \
+        port:py${python.version}-six
+
+    if { ${python.version} < 35 } {
+        depends_lib-append \
+            port:py${python.version}-typing \
+    }
+
+    if { ${python.version} >= 37 } {
+        depends_lib-append \
+            port:py${python.version}-typing_extensions
+    }
+
+    depends_test-append \
+        port:py${python.version}-nbval \
+        port:py${python.version}-pytest \
+        port:py${python.version}-tabulate
+
+    post-fetch {
+        system -W ${worksrcpath} "git submodule update --init --recursive"
+    }
+
+    test.run        yes
+    test.cmd        py.test-${python.branch}
+    test.target     onnx/test
+    test.env        PYTHONPATH=${worksrcpath}/build/lib
+
```

Comment:
again, it doesn't look like you actually run the tests with (```sudo port -dv test py37-onnx```) as you say you did in the PR template. Or you did only so after installing the port first; at least when directly running the tests without first installing the port (which should work) you get errors like: ```E   ModuleNotFoundError: No module named 'onnx'```

That's because it compiles the extension and, therefore, your setting of PYTHONPATH is incorrect. You should use
``` 
    pre-test {
        test.env    PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]
    }
```

Reply:
Thanks done.